[Ren Guanpeng] iP

GuanpengR opened the pull request:
### MGTOW

> "The secret of getting ahead is getting started." – Mark Twain

MGTOW helps you get ahead by managing your tasks efficiently. It's,

-  command-line based
-  intuitive
-  _SUPER_ FAST to use

All you need to do is,
1. download it from [here](https://github.com/GuanpengR/ip/releases/download/A-Jar/mgtow.jar).
2. open a command window in that folder.
3. Run the command `java -jar "Mgtow.jar"`
4. start managing your tasks with ease! 🚀

And it is **FREE**!

Features:

- [x]  Managing various types of tasks
  - To-Do tasks
  - Deadlines
  - Events
- [x] Finding tasks by keyword
- [x] Marking tasks as done/undone
- [ ] Reminders (coming soon)

## Coming Soon:
* GUI interface
* Task prioritization
* Recurring tasks

If you're a Java programmer, you can use it to practice Java too. Here's the `main` method to get you started:

```java
public class Mgtow {
    public static void main(String[] args) {
        new Mgtow("./data/mgtow.MGTOW.txt").run();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

GuanpengR committed:
Rename Duke to MGTOW, added greet and exit messages, implemented echo user input and exit command "bye"

GuanpengR committed:
Created classes Task, TaskStorage, ToDo, Deadline, Events and error handling as well as Deletion

GuanpengR committed:
Add load and save tasks

GuanpengR committed:
add date time interpretation and find tasks by date

GuanpengR committed:
Use more OOP

GuanpengR committed:
Merge branch 'add-gradle-support'

GuanpengR committed:
Organize into packages and add JUnit tests

GuanpengR committed:
Update build.gradle

GuanpengR committed:
Add JavaDoc

GuanpengR committed:
Edit to follow coding standard

GuanpengR committed:
Merge branch 'branch-A-CodingStandard'

GuanpengR committed:
Add search by keyword

GuanpengR committed:
Merge branch 'branch-Level-9'

fusin02 commented on src/main/java/mgtow/Mgtow.java:
```diff
@@ -0,0 +1,102 @@
+import java.util.ArrayList;
+
+/**
+ * Main class for the MGTOW (Man Going Their Own Way) task management application.
```
Love this name!

fusin02 commented on src/main/java/mgtow/Mgtow.java:
```diff
@@ -0,0 +1,102 @@
+                String[] commandParts = Parser.parseCommand(fullCommand);
+                switch (commandParts[0]) {
+                    case "bye":
+                        isFinished = true;
```
Good naming of boolean! Clearly shows what this variable is flagging

fusin02 commented on src/main/java/mgtow/storage/Storage.java:
```diff
@@ -0,0 +1,58 @@
+        createDirectoryIfNotExists();
+    }
+
+    private void createDirectoryIfNotExists() {
```
Good method naming, clearly summarises method

fusin02 commented on src/main/java/mgtow/ui/Ui.java:
```diff
@@ -0,0 +1,84 @@
+ * Manages the Ui in the Mgtow application.
+ * This class provides methods for displaying responses to the user.
+ */
+public class Ui {
```
Methods are all appropriately named as verbs and all are clear, good job bro !

fusin02 commented on src/main/java/mgtow/task/TaskList.java:
```diff
@@ -0,0 +1,116 @@
+ * This class provides methods for adding, deleting, and manipulating tasks.
+ */
+public class TaskList {
+    private ArrayList<Task> tasks;
```
Good use of plurals to make it clear that tasks is a collection of tasks.

fusin02 reviewed (commented):
Great job with the coding quality and naming!

GuanpengR commented:
> Great job with the coding quality and naming!

Thank you!

GuanpengR committed:
Add GUI to Mgtow

jtooya commented on src/main/java/mgtow/Mgtow.java:
```diff
@@ -0,0 +1,103 @@
+                ui.showLine();
+                String response = processCommand(fullCommand);
+                ui.showMessage(response);
+                if (fullCommand.equals("bye")) {
```
Should this take into consideration the capitalisation of the "bye" message? For example, if the user inputs "Bye", this would not evaluate to true

jtooya commented on src/main/java/mgtow/storage/Storage.java:
```diff
@@ -0,0 +1,58 @@
+import mgtow.util.InvalidTaskException;
+import mgtow.util.Parser;
+
+import java.io.*;
```
Perhaps we can consider importing each method explicitly to keep the amount of code imported to a minimum?

jtooya commented on src/main/java/mgtow/storage/Storage.java:
```diff
@@ -0,0 +1,58 @@
+        }
+    }
+
+    public ArrayList<Task> load() throws InvalidTaskException {
```
Could we consider changing the name to be more specific? Like "loadTasks()" or "loadTasksFromStorage()"?

jtooya commented on src/main/java/mgtow/task/Task.java:
```diff
@@ -0,0 +1,64 @@
+
+    @Override
+    public String toString(){
+        return String.format("[%s][%s] %s", this.type, this.getStatus(), this.desc);
```
I like the cleanliness of this bit of code!

jtooya commented on src/main/java/mgtow/ui/Ui.java:
```diff
@@ -0,0 +1,100 @@
+        return "OK, I've unmarked this task:\n  " + task;
+    }
+
+    public String getDeleteTaskMessage(Task task, int remainingTasks) {
+        return "Noted. I've removed this task:\n  " + task +
+                "\nNow you have " + remainingTasks + " tasks in the list.";
+    }
+
+    public String getAddTaskMessage(Task task, int totalTasks) {
+        return "Got it. I've added this task:\n  " + task +
+                "\nNow you have " + totalTasks + " tasks in the list.";
+    }
```
This is great distinction of parameter names to not get the two methods confused!

GuanpengR committed:
Add assertions

Add assertions to ensure valid dates
Add assertions to check creation of new object is performed correctly

GuanpengR committed:
Improve code quality

Rename method to be more specific
Import all classes explicitly

GuanpengR committed:
Merge branch 'branch-A-CodeQuality'

GuanpengR committed:
Add sort by date

Add sort by date to display tasks by date
Todos are at sorted at the end as they do not have deadline

GuanpengR committed:
Delete mgtow.MGTOW.txt

GuanpengR committed:
Merge pull request #1 from GuanpengR/branch-C-Sort

Delete mgtow.MGTOW.txt

GuanpengR committed:
Add javadoc

GuanpengR committed:
Add user guide

GuanpengR committed:
Add exit on bye command

GuanpengR committed:
Update build.gradle
